cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -1406,6 +1458,9 @@ void TestBackpressure(BatchesMaker maker, int num_batches, int batch_size,
     total_length += batch->num_rows();
   }
   ASSERT_EQ(static_cast<int64_t>(num_batches * batch_size), total_length);
+
+  ASSERT_GT(pause_count, 0);
```

Comment:
Can we validate pause resume counter for all sources? i.e the slow table should not have been paused, but the fast tables should?

Reply:
We may have crossed - the [previous commit](https://github.com/apache/arrow/pull/35874/commits/7d40e92c7d96e34935b2af3f84f74d4e1d8cec49) already has one pause and one resume counter for each source. The issue is that sometimes (due to non-deterministic timing of operations) no pause/resume is requested on one or both of the fast sources. Because of this, in the [recent commit](https://github.com/apache/arrow/pull/35874/commits/3e5fdf9a4a7e48500882f1592d1949c772dca358) I am attempting this test logic: there is a pause request on a fast source if and only if there is a resume one too.